### ISSUE TITLE
hfst-sfstpl2fst fails with SFSTs with variable definition

### ISSUE THREAD

**Issue by santhoshtr** [hfst/hfst#369]
This is my test fst in SFST formalism. Verbatim copy of example from SFST manual , variables introduction.

```
$Nsg$ = house | foot | mouse
$Npl$ = house<>:s | f o:e o:e t | {mouse}:{mice}
$Nsg$ <N>:<> <sg>:<> | $Npl$ <N>:<> <pl>:<>
```
Trying to compile:
```
practice$ fst-compiler-utf8  variables.fst variables.o                     
variables.fst: 4
```
As you can see it compiles perfectly.

```
practice$ fst-compiler-utf8 -v
fst-compiler version 1.4.7a
```
Version is 1.4.7a. Note that OS is Ubuntu 17.04.

Now using hfst-sfstpl2fst 

```
practice$ hfst-sfstpl2fst -i variables.fst -o variables.o
undefined variable $Nsg$

Exception: HfstException in file: SfstCompiler.cc on line: 282

practice$ hfst-sfstpl2fst -V
hfst-sfstpl2fst 0.1 (hfst 3.13.0)
Copyright (C) 2017 University of Helsinki,
License GPLv3: GNU GPL version 3 <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
practice$ 
```

Error is "Exception: HfstException in file: SfstCompiler.cc on line: 282". This happens with any sfst with variable definitions.

**Comment by santhoshtr** [comment]
This issue happens only when I compile and use hfst from the released tags. WIth hfst in ubuntu 17.04, all works fine with no issues. That is hfst 3.9.0 though.

**Comment by santhoshtr** [comment]
I bisected till 5bcb93dbe33674c0cdcce225a04d4d8b4acef694
All works till there, may be a few more commits. 
+cc @eaxelson 
I think this is a serious regression for SFST support.

**Comment by eaxelson** [comment]
Now using strings instead of char pointers in variable maps (commit 8251357), does it work? This is a quick fix, the whole variable handling and unordered containers should be rewritten...

**Comment by santhoshtr** [comment]
Not really. I compiled the master version with the above commit and tried compiling my SFST files. Some are passing, but many failing. Here is an example failure:
```
ninfl (master)$ hfst-sfstpl2fst -f openfst-tropical -v -i accusative.fst -o accusative.a
accusative.fst: 5
  ../symbols.fst: 54
accusative.fst: 16terminate called after throwing an instance of 'std::logic_error'
  what():  basic_string::_M_construct null not valid
Aborted (core dumped)
```
(the file is at https://github.com/smc/mlmorph/tree/master/ninfl )

Also compiling lexicon is tooo slow compared to sfst-compile-utf8 for https://github.com/smc/mlmorph/tree/master/lexicon

While we are fixing this, can I request some automatic tests in hfst for SFST PL with some medium sophisticated code?
